enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/CatchUpClient.java

```diff
@@ -172,11 +172,25 @@ public boolean isActive()
         }
 
         @Override
-        public void close()
+        public CompletableFuture<Void> close()
         {
             if ( nettyChannel != null )
             {
-                nettyChannel.close();
+                return CompletableFuture.runAsync( () ->
+                {
+                    try
+                    {
+                        nettyChannel.close().sync();
```

Comment:
It would probably be nicer to convert from Netty `Future` to `CompletableFuture` by creating and incomplete `CompletableFuture` and completing it from a `Future` listener. Otherwise this will block a thread from common pool for unknown amount of time.